editoast/editoast_models/src/rolling_stock/rolling_stock_category.rs

```diff
@@ -0,0 +1,84 @@
+use std::io::Write;
+use std::str::FromStr;
+
+use diesel::deserialize::FromSql;
+use diesel::deserialize::FromSqlRow;
+use diesel::expression::AsExpression;
+use diesel::pg::Pg;
+use diesel::pg::PgValue;
+use diesel::serialize::Output;
+use diesel::serialize::ToSql;
+use serde::Deserialize;
+use serde::Serialize;
+use strum::Display;
+use strum::EnumString;
+use strum::IntoStaticStr;
+use utoipa::ToSchema;
+
+editoast_common::schemas! {
+    RollingStockCategory,
+    RollingStockCategories,
+}
+
+// This enum maps to a Postgres enum type, specifically `rolling_stock_category`.
+// Any changes made to this enum must be reflected in the corresponding Postgres enum,
+// and vice versa, to ensure consistency between the application and the database.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Serialize,
+    Deserialize,
+    ToSchema,
+    FromSqlRow,
+    AsExpression,
+    EnumString,
+    IntoStaticStr,
+    Display,
+)]
+#[diesel(sql_type = crate::tables::sql_types::RollingStockCategory)]
+#[strum(serialize_all = "kebab-case")]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum RollingStockCategory {
```

Comment:
@leovalais Why this enum is defined in the model and not in the `osrd-schema`.

I don't see how different this enum is from `LoadingGaugeType`. The enum should be defined in the schema and used in models with the annotation `to_enum`.

Reply:
Agreed that this enum should be defined in `editoast-schemas`.

Disagreed about enums being stored as integers instead of PG enums (cc @woshilapin who suggested this). I believe we should allow a `#[model(enum = <type>)]` where `type` is either `u8/16/64` or `&str` by default. (We're not proposing to store variants as strings in db, we just convert them to string to write the query.)

Reply:
If we move the categories enum into `editoast-schemas`, we need to add `diesel` as a dependency in `editoast-schemas`, and I don't think it's desirable. what do you think?

Reply:
You're right we should try to keep `diesel` only in `editoast_models`, you'll need an `impl From<editoast_schemas::RSCategory> for editoast_models::RsCategory`

Reply:
If I understand correctly, we need to define another enum for categories in `editoast_schemas`. If so, why is it necessary since the enum is already in `editoast_models`?

Reply:
Because `editoast-schemas` aims to be the Rust counterpart of `osrd-schemas`

Reply:
So, I need to define another enum in `editoast-schemas`, right?

Reply:
Since the enum is now defined in `osrd-schemas` we could do something like this to avoid duplication:

```suggestion
use osrd_schemas::rolling_stock::RollingStockCategory as RollingStockCategorySchema;
pub struct RollingStockCategory(RollingStockCategorySchema);
```

Reply:
Done.